contracts/mixnet/src/epoch/queries.rs

```diff
@@ -0,0 +1,422 @@
+// Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use super::storage;
+use crate::error::ContractError;
+use cosmwasm_std::{Env, Order, StdResult, Storage};
+use cw_storage_plus::Bound;
+use mixnet_contract_common::{
+    Epoch, EpochRewardedSetHeightsResponse, IdentityKey, PagedRewardedSetResponse,
+    RewardedSetNodeStatus, RewardedSetUpdateDetails,
+};
+
+pub fn query_current_epoch(storage: &dyn Storage) -> Result<Epoch, ContractError> {
+    Ok(storage::CURRENT_EPOCH.load(storage)?)
+}
+
+pub(crate) fn query_rewarded_set_refresh_minimum_blocks() -> u32 {
+    crate::contract::REWARDED_SET_REFRESH_BLOCKS
+}
+
+pub fn query_rewarded_set_heights_for_epoch(
+    storage: &dyn Storage,
+    epoch_id: u32,
+) -> Result<EpochRewardedSetHeightsResponse, ContractError> {
+    // I don't think we have to deal with paging here as at most we're going to have 720 values here
+    // and I think the validators are capable of performing 720 storage reads at once if they're only
+    // reading u64 (+ u8) values...
+    let heights = storage::REWARDED_SET_HEIGHTS_FOR_EPOCH
+        .prefix(epoch_id)
+        .range(storage, None, None, Order::Ascending)
+        .map(|val| val.map(|(height, _)| height))
+        .collect::<StdResult<Vec<_>>>()?;
+
+    Ok(EpochRewardedSetHeightsResponse { epoch_id, heights })
+}
+
+// note: I have removed the `query_rewarded_set_for_epoch`, because I don't think it's appropriate
+// for the contract to go through so much data (i.e. all "rewarded" sets of particular epoch) in one go.
+// To achieve the same result, the client would have to instead first call `query_rewarded_set_heights_for_epoch`
+// to learn the heights used in given epoch and then for each of them `query_rewarded_set` for that particular height.
+
+pub fn query_current_rewarded_set_height(storage: &dyn Storage) -> Result<u64, ContractError> {
+    Ok(storage::CURRENT_REWARDED_SET_HEIGHT.load(storage)?)
+}
+
+fn query_rewarded_set_at_height(
+    storage: &dyn Storage,
+    height: u64,
+    start_after: Option<IdentityKey>,
+    limit: u32,
```

Comment:
Should there be a default limit  implementation here, like we have below?

Reply:
So this function is kinda an artifact of experimentation with different kinds of queries and trying to extract common code. If you notice, this function is not public and is only used internally by `query_rewarded_set`. So I guess the question is: should those be combined together or left separated as they are now

Reply:
I'm ok with it being separate, I'm thinking that someone will start using it in a different place, and then the lack of a default limit will bite us :)